### PR TITLE
HMAN-34 add Pact tests

### PR DESCRIPTION
**Pull request by markdathornehmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-34

### Change description ###
PACT Description for HMC Endpoint - GET hearings

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -295,6 +349,20 @@ dependencies {
...
+
+  contractTestCompile group: 'au.com.dius.pact.consumer', name: 'junit5', version: versions.pact
+  contractTestRuntime group: 'au.com.dius.pact.consumer', name: 'junit5', version: versions.pact
+
+  contractTestCompile group: 'au.com.dius.pact.provider', name: 'junit5', version: versions.pact
+  contractTestCompile group: 'au.com.dius.pact.provider', name: 'spring', version: versions.pact
+  contractTestCompile group: 'au.com.dius.pact.provider', name: 'junit5spring', version: versions.pact
+
+  contractTestImplementation group: 'org.projectlombok', name: 'lombok', version: versions.lombok
+  annotationProcessor group: 'org.projectlombok', name: 'lombok', version: versions.lombok
```
Comment (RebeccaBaker) [inline build.gradle:364]: already imported annotationProcessor for lombok on line 342

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java [hunk 2]
```diff
@@ -62,7 +63,7 @@ public void invokeHearing(@RequestBody @Valid HearingRequest hearingRequest) {
         @ApiResponse(code = 200, message = "Success (with content)"),
         @ApiResponse(code = 400, message = "Invalid request")
     })
-    public HearingRequest getHearingsRequest(@PathVariable("ccdCaseRef")
+    public HearingsGetResponse getHearingsRequest(@PathVariable("ccdCaseRef")
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java:66]: why has the response changed? To me this is less clear

File: src/main/java/uk/gov/hmcts/reform/hmc/model/HearingCreatePostResponse.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    @NotEmpty(message = ValidationError.HEARING_STATUS_NULL_EMPTY_INVALID)
+    @Pattern(regexp = "^\\w{100}$", message = ValidationError.HEARING_STATUS_MAX_LENGTH_EMPTY_INVALID)
+    private String status;
+
+    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
+    @JsonSerialize(using = LocalDateTimeSerializer.class)
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
+    private LocalDateTime timeStamp;
+}
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/model/HearingCreatePostResponse.java:34]: This class is only used for pact testing - should be storing testing model classes in the main java level?

File: src/main/java/uk/gov/hmcts/reform/hmc/model/HearingWindow.java [hunk 4]
```diff
@@ -3,16 +3,11 @@
...
-import java.time.LocalDateTime;
-
 @Data
 @NoArgsConstructor
 public class HearingWindow {
 
-    private LocalDate hearingWindowStartDateRange;
-
-    private LocalDate hearingWindowEndDateRange;
+    private HearingWindowDateRange hearingWindowDateRange;
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/model/HearingWindow.java:10]: were these refactors part of the ticket and scope?

**Review by RebeccaBaker — COMMENTED** [review]
some minor comments

File: build.gradle (hunk 1 above)
Comment (markdathornehmcts) [reply build.gradle]: Indeed. Corrected

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java (hunk 2 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementController.java:66]: Dan pointed out that the response wasn't correct. The response should have been a limited selection of fields.  In hindsight, this should have been address in a separate JIRA.

File: src/main/java/uk/gov/hmcts/reform/hmc/model/HearingWindow.java (hunk 4 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/model/HearingWindow.java:10]: No. Should have been a separate ticket.

File: src/main/java/uk/gov/hmcts/reform/hmc/model/HearingCreatePostResponse.java (hunk 3 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/model/HearingCreatePostResponse.java:34]: Part of out of scope fix. Will discuss best way to address. Probably needs a separate Jira.

**Pull request closed without merging**